src/librustc_index/vec.rs

```diff
@@ -669,6 +669,17 @@ impl<I: Idx, T> IndexVec<I, T> {
         }
     }
 
+    /// Returns mutable references to three distinct elements or panics otherwise.
+    #[inline]
+    pub fn pick3_mut(&mut self, a: I, b: I, c: I) -> (&mut T, &mut T, &mut T) {
+        let (ai, bi, ci) = (a.index(), b.index(), c.index());
+        assert!(ai != bi && bi != ci && ci != ai);
+        let len = self.raw.len();
+        assert!(ai < len && bi < len && ci < len);
+        let ptr = self.raw.as_mut_ptr();
+        unsafe { (&mut *ptr.add(ai), &mut *ptr.add(bi), &mut *ptr.add(ci)) }
```

Comment:
you can also write this in safe code by using `split_at_mut` twice

Reply:
I might be missing something, but it seems like it would require some non-trivial amount of code, because we don't know relative order of indices, but need to preserve the order, so there are six different cases to consider.

Reply:
ah... that's a good point indeed. I thought they were ordered